extra_tests/snippets/aot.py

```
@@ -0,0 +1,186 @@
+assert untyped("a", "b") == "ab"
+assert untyped(1, 2) == 3
+assert variadic(1, 2, 3) == 6
+assert closure_factory(5)(1) == 6
+assert list(generator(7)) == [7]
+assert guarded(3) == 3
+
+# Division by zero raises rather than returning inf or killing the process.
+try:
+    scale(1.0, 0.0)
+except ZeroDivisionError:
+    raise AssertionError("scale does not divide")
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)